astropy/coordinates/sky_coordinate.py

```diff
@@ -72,9 +72,11 @@ def _represent_as_dict(self):
         if 'distance' in attrs and np.all(obj.distance == 1.0):
             attrs.remove('distance')
 
-        self._represent_as_dict_attrs = attrs
-
-        out = super()._represent_as_dict()
+        out = {}
```

Comment:
Use `data_info._get_obj_attrs_map`.  (Which maybe needs to be made public?).

Or maybe `_represent_as_dict` could take an optional `attrs` to allow overriding `self._represent_as_dict_attrs`?

Reply:
Much better. Here, `represent_as_dict` itself is overridden, but, yes, a `super().represent_as_dict(attrs)` would work. I'm not quite sure it is worth changing, especially if we're considering a larger overhaul.

Reply:
Ah, darn, I now see why you suggested it: `_get_obj_attrs_map` is not on `DataInfo` but in the `data_info` module...